resourcet/Data/Acquire/Internal.hs

```diff
@@ -25,8 +27,21 @@ import qualified Control.Monad.Catch as C ()
 -- @since 1.1.2
 data ReleaseType = ReleaseEarly
                  | ReleaseNormal
-                 | ReleaseException
-    deriving (Show, Read, Eq, Ord, Enum, Bounded, Typeable)
+                 | ReleaseException' E.SomeException
+    deriving (Show, Typeable)
+
+-- | Fake 'E.Exception' to use with the deprecated 'ReleaseException' pattern.
+data DeprecatedReleaseExceptionPlaceholder = DeprecatedReleaseExceptionPlaceholder
+     deriving (Show)
+
+instance E.Exception DeprecatedReleaseExceptionPlaceholder
+
+{-# COMPLETE ReleaseEarly, ReleaseNormal, ReleaseException #-}
```

Comment:
This `COMPLETE` pragma means that GHC will not issue a warning when you do the following:

```haskell
case x of
    ReleaseEarly -> print 1
    ReleaseNormal -> print 2
    ReleaseException -> print 3
```

Without the pragma, GHC has no way of knowing that `ReleaseException` handles the `ReleaseException'` case, and will complain that you aren't handling `ReleaseException' _`.

GHC does not check this pragma - it trusts it.